designer/server/src/plugins/errorPage.ts

```diff
@@ -6,6 +6,12 @@ import {
 
 import { errorViewModel } from '~/src/models/errors.js'
 
+const errorCodes: Record<number, string> = {
+  403: 'You do not have access to this service',
+  404: 'Page not found',
+  500: 'Sorry, there is a problem with the service'
+}
```

Comment:
Best dropping the types or using `satisfies` otherwise the compiler will think all numbers are in here

```suggestion
const errorCodes = {
  403: 'You do not have access to this service',
  404: 'Page not found',
  500: 'Sorry, there is a problem with the service'
} satisfies Record<number, string>
```

Reply:
Sorry, this is my fault, TypeScript is going to treat these as string properties

You're best using a map for "does this thing exist"

```js
const errorCodes = new Map([
  [403, 'You do not have access to this service'],
  [404, 'Page not found'],
  [500, 'Sorry, there is a problem with the service']
])
```